### PR TITLE
Add cura stl thumbnails

### PR DESCRIPTION
**Pull request by mobilegmYT**
Installs https://github.com/unlimitedbacon/stl-thumb which adds thumbnails to stls in file manager.

### CODE REVIEW

**Comment by Itai-Nelken** [comment]
Will wait for @Botspot for this one

**Comment by Botspot** [comment]
Three things I notice:

1. I don't believe the 3rd-party Cura appimage **accepts arguments** for files to open. This fact makes any filetypes or icons fairly useless because you wouldn't be able to double-click a STL file to open it in Cura.
2. The script does not delete the deb file after installing it.
3. What if the user already had `stl-thumb` installed prior to this? (Hint: use `pkg-install` to install the local deb)

**Comment by mobilegmYT** [comment]
> 3. What if the user already had `stl-thumb` installed prior to this? (Hint: use `pkg-install` to install the local deb)
Can pkg-install parse deb files?

**Comment by Itai-Nelken** [comment]
> Can pkg-install parse deb files?

Yes

**Comment by Itai-Nelken** [comment]
> Can pkg-install parse deb files?

Yes

File: apps/Cura/install-64 [hunk 1]
```diff
@@ -7,6 +7,11 @@ function error {
   exit 1
 }
 
+wget https://github.com/unlimitedbacon/stl-thumb/releases/download/v0.4.0/stl-thumb_0.4.0_arm64.deb
+
+"${DIRECTORY}/pkg-install" "./stl-thumb_0.4.0_arm64.deb" "$(dirname "$0")" || exit 1
```
Comment (Itai-Nelken) [inline apps/Cura/install-64:12]:
```suggestion
"${DIRECTORY}/pkg-install" "~/stl-thumb_0.4.0_arm64.deb" "$(dirname "$0")" || exit 1
```

File: apps/Cura/install-32 [hunk 2]
```diff
@@ -7,6 +7,9 @@ function error {
   exit 1
 }
 
+wget https://github.com/unlimitedbacon/stl-thumb/releases/download/v0.4.0/stl-thumb_0.4.0_armhf.deb
+sudo apt install -y ./stl-thumb_0.4.0_armhf.deb
```
Comment (Itai-Nelken) [inline apps/Cura/install-32:11]: Use pkg-install here as well

**Review by Itai-Nelken — CHANGES_REQUESTED** [review]

**Comment by Botspot** [comment]
@Itai-Nelken, I'm not sure that `pkg-install` is capable of parsing input files beginning with `./`. Was your change tested?

More importantly, I consider this PR rather useless as long as Cura cannot accept file inputs.

**Comment by mobilegmYT** [comment]
Closin then.

**Comment by Botspot** [comment]
> as long as Cura cannot accept file inputs.

Oh wait, it looks like Cura had an update and now it does accept file arguments.

**Comment by mobilegmYT** [comment]
> > as long as Cura cannot accept file inputs.
> 
> Oh wait, it looks like Cura had an update and now it does accept file arguments.

*wheeze*

**Comment by Botspot** [comment]
Looks like the Cura desktop file needs modifying to work. (inserting a %F somewhere)

**Comment by Botspot** [comment]
> Looks like the Cura desktop file needs modifying to work. (inserting a %F somewhere)

Finally.
```
Exec=bash -c 'sudo modprobe fuse; ~/Cura.AppImage '\ "%f"
```
This was roughly my 25th attempt. All previous attempts either failed to open the file, failed to open Cura, or failed to work for filenames containing a space character.

**Comment by Itai-Nelken** [comment]
>@Itai-Nelken, I'm not sure that pkg-install is capable of parsing input files beginning with ./. Was your change tested?

`$(pwd)` works or the full path also works. I think that `~/` also works.

**Comment by Botspot** [comment]
d105473d9f5d95f6e3f1e737cacdd6a5a10e3f50

**Pull request closed without merging**